grpc-bridge/src/main/java/org/jboss/resteasy/grpc/servlet/HttpServletRequestImpl.java

```diff
@@ -0,0 +1,670 @@
+package org.jboss.resteasy.grpc.servlet;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.jboss.resteasy.grpc.i18n.Messages;
+import org.jboss.resteasy.grpc.util.DateUtils;
+import org.jboss.resteasy.grpc.util.IteratorEnumeration;
+import org.jboss.resteasy.grpc.util.LocaleUtils;
+import org.jboss.resteasy.specimpl.ResteasyUriInfo;
+
+import jakarta.servlet.AsyncContext;
+import jakarta.servlet.DispatcherType;
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletInputStream;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletRequestWrapper;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.ServletResponseWrapper;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import jakarta.servlet.http.HttpUpgradeHandler;
+import jakarta.servlet.http.Part;
+import jakarta.ws.rs.NotSupportedException;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.UriInfo;
+
+/*
+ * The http servlet request implementation. This class is not thread safe.
+ *
+ * Much of it is borrowed from io.undertow.servlet.spec.HttpServletRequestImpl,
+ * written by Stuart Douglas.
+ */
+public class HttpServletRequestImpl implements HttpServletRequest {
```

Comment:
I need to spend some time to grok this, but it seems odd we'd have to fully implement an `HttpServletRequest` without a delegate.

Reply:
We're coming in through the gRPC runtime, so there's no HttpServletRequest to delegate to. I looked for a way to hook into undertow to create one of its HttpServletRequests, but I didn't find a way to do it.